### PR TITLE
remove VisualFSharpOpenSource package

### LINKED ISSUE #4496: Can't install/use VisualFSharpOpenSource.vsix
Doing this:

    .\build.cmd vs
    VSIXInstaller.exe release\net40\bin\VisualFSharpOpenSource.vsix

I get this error 

image

    12/03/2018 17:42:15 - Install Error : System.InvalidOperationException: A value for 'Component' needs to be specified in the catalog.

Here is the end of the install log, [full log is here](https://gist.github.com/dsyme/ff483b70867ac307a63dd7bef4884012)

```
12/03/2018 17:42:13 - Upgrading 'Visual F# Tools', version 15.6.0.18021307 to version 42.42.42.42.
12/03/2018 17:42:15 - Install Error : System.InvalidOperationException: A value for 'Component' needs to be specified in the catalog.
   at Microsoft.VisualStudio.ExtensionManager.PackageInstaller.LoadComponent(IDependencyGraph dGraph, Boolean& isProductComponent)
   at Microsoft.VisualStudio.ExtensionManager.ExtensionEngineImpl.PerformSetupEngineInstall(InstallableExtensionImpl extension, Boolean installPerMachine, Boolean isPackComponent, IDictionary`2 extensionsInstalledSoFar, List`1 extensionsUninstalledSoFar, IInstalledExtensionList modifiedInstalledExtensionsList, IProgress`1 progress, InstallFlags installFlags, AsyncOperation asyncOp, Version targetedVsVersion, IInstalledExtension& newExtension)
   at Microsoft.VisualStudio.ExtensionManager.ExtensionEngineImpl.InstallInternal(InstallableExtensionImpl extension, InstallFlags installFlags, IDictionary`2 extensionsInstalledSoFar, List`1 extensionsUninstalledSoFar, IInstalledExtensionList modifiedInstalledExtensionsList, AsyncOperation asyncOp, IProgress`1 progress, Version targetedVsVersion)
   at Microsoft.VisualStudio.ExtensionManager.ExtensionEngineImpl.BeginInstall(IInstallableExtension installableExtension, InstallFlags installFlags, AsyncOperation asyncOp, Version targetedVsVersion)
   at Microsoft.VisualStudio.ExtensionManager.ExtensionEngineImpl.InstallWorker(IInstallableExtension extension, InstallFlags installFlags, AsyncOperation asyncOp)
```

#### Expected behavior

Can install and use open source build

#### Related information 

Provide any related information 

* Visual Studio 15.6.0
* master

### PR DESCRIPTION
**Pull request by brettfo**
At this point the only differences between `VisualFSharpOpenSource` and `VisualFSharpFull` are (1) the package ID, (2) the inclusion of the project templates, and (3) the version.  All three of those points are now addressable:

1.  The differing package IDs causes problems when trying to run `VSIXInstaller.exe VisualFSharpOpenSource.vsix` so directly using `VisualFSharpFull.vsix` is a better solution anyways.
2.  The project templates were added to `VisualFSharpOpenSource` as a stop-gap fix for when project templates were factored out into a separate package, but the shipped version of Visual Studio still depended on those templates being in that package.  Since VS 15.5 and newer this is no longer the case.
3.  We dynamically calculate the 'real' version of the VSIX package based on the build number or date and the version of `VisualFSharpOpenSource` was hard-coded to `42.42.42.42` for two reasons, (a) it will always be larger than `15.x` of the installed package, and (b) it's obviously contrived so there's no question as to if it's official or not.  In this change `42.42.42.42` is returned if we're not in a CI-type environment, e.g., a developer's desktop machine.

With these changes I can both F5 the `VisualFSharpFull` project and run `VSIXInstaller.exe VisualFSharpFull.vsix`.

Fixes #4496.

### CODE REVIEW

**Comment by brettfo** [comment]
@dotnet-bot test this

**Comment by brettfo** [comment]
Jenkins runs aren't getting started, but they were for PRs created a few hours ago.  @mmitche are there known Jenkins outages?

**Comment by mmitche** [comment]
@brettfo There was some misconfiguration in azure that should be fixed.

@dotnet-bot test this please

File: vsintegration/update-vsintegration.cmd [hunk 1]
```diff
@@ -117,12 +117,12 @@ echo.
...
 echo.
 echo ^> VSIXInstaller.exe /u:"VisualFSharp"
-echo ^> VSIXInstaller.exe release\net40\bin\VisualFSharpOpenSource.vsix
+echo ^> VSIXInstaller.exe release\net40\bin\VisualFSharpFull.vsix
 echo.
 echo For Debug builds:
 echo.
 echo ^> VSIXInstaller.exe /u:"VisualFSharp"
-echo ^> VSIXInstaller.exe debug\net40\bin\VisualFSharpOpenSource.vsix
+echo ^> VSIXInstaller.exe debug\net40\bin\VisualFSharpFull.vsix
```
Comment (dsyme) [inline vsintegration/update-vsintegration.cmd]: Maybe just change the name of the package to VisualFSharp.vsix

**Comment by AviAvni** [comment]
running build.cmd vs on latest master I get error

```
"C:\Users\aviav\Source\Repos\visualfsharp\build-everything.proj" (default target) (1) ->
"C:\Users\aviav\Source\Repos\visualfsharp\vsintegration\fsharp-vsintegration-vsix-build.proj" (Build target) (32) ->
"C:\Users\aviav\Source\Repos\visualfsharp\vsintegration\Vsix\VisualFSharpFull\VisualFSharpFull.csproj" (Build target) (
33) ->
(GenerateTemplatesManifest target) ->
  C:\Users\aviav\Source\Repos\visualfsharp\Packages\Microsoft.VSSDK.BuildTools\15.1.192\tools\vssdk\Microsoft.VsSDK.tar
gets(580,5): error MSB4018: The "CreateTemplatesManifest" task failed unexpectedly. [C:\Users\aviav\Source\Repos\visual
fsharp\vsintegration\Vsix\VisualFSharpFull\VisualFSharpFull.csproj]
C:\Users\aviav\Source\Repos\visualfsharp\Packages\Microsoft.VSSDK.BuildTools\15.1.192\tools\vssdk\Microsoft.VsSDK.targe
ts(580,5): error MSB4018: System.IO.InvalidDataException: End of Central Directory record could not be found. [C:\Users
\aviav\Source\Repos\visualfsharp\vsintegration\Vsix\VisualFSharpFull\VisualFSharpFull.csproj]
C:\Users\aviav\Source\Repos\visualfsharp\Packages\Microsoft.VSSDK.BuildTools\15.1.192\tools\vssdk\Microsoft.VsSDK.targe
ts(580,5): error MSB4018:    at System.IO.Compression.ZipArchive.ReadEndOfCentralDirectory() [C:\Users\aviav\Source\Rep
os\visualfsharp\vsintegration\Vsix\VisualFSharpFull\VisualFSharpFull.csproj]
C:\Users\aviav\Source\Repos\visualfsharp\Packages\Microsoft.VSSDK.BuildTools\15.1.192\tools\vssdk\Microsoft.VsSDK.targe
ts(580,5): error MSB4018:    at System.IO.Compression.ZipArchive.Init(Stream stream, ZipArchiveMode mode, Boolean leave
Open) [C:\Users\aviav\Source\Repos\visualfsharp\vsintegration\Vsix\VisualFSharpFull\VisualFSharpFull.csproj]
C:\Users\aviav\Source\Repos\visualfsharp\Packages\Microsoft.VSSDK.BuildTools\15.1.192\tools\vssdk\Microsoft.VsSDK.targe
ts(580,5): error MSB4018:    at System.IO.Compression.ZipArchive..ctor(Stream stream, ZipArchiveMode mode, Boolean leav
eOpen, Encoding entryNameEncoding) [C:\Users\aviav\Source\Repos\visualfsharp\vsintegration\Vsix\VisualFSharpFull\Visual
FSharpFull.csproj]
C:\Users\aviav\Source\Repos\visualfsharp\Packages\Microsoft.VSSDK.BuildTools\15.1.192\tools\vssdk\Microsoft.VsSDK.targe
ts(580,5): error MSB4018:    at System.IO.Compression.ZipFile.Open(String archiveFileName, ZipArchiveMode mode, Encodin
g entryNameEncoding) [C:\Users\aviav\Source\Repos\visualfsharp\vsintegration\Vsix\VisualFSharpFull\VisualFSharpFull.csp
roj]
C:\Users\aviav\Source\Repos\visualfsharp\Packages\Microsoft.VSSDK.BuildTools\15.1.192\tools\vssdk\Microsoft.VsSDK.targe
ts(580,5): error MSB4018:    at Microsoft.VisualStudio.Sdk.BuildTasks.CreateTemplatesManifest.UnzipFilesInSource() [C:\
Users\aviav\Source\Repos\visualfsharp\vsintegration\Vsix\VisualFSharpFull\VisualFSharpFull.csproj]
C:\Users\aviav\Source\Repos\visualfsharp\Packages\Microsoft.VSSDK.BuildTools\15.1.192\tools\vssdk\Microsoft.VsSDK.targe
ts(580,5): error MSB4018:    at Microsoft.VisualStudio.Sdk.BuildTasks.CreateTemplatesManifest.Execute() [C:\Users\aviav
\Source\Repos\visualfsharp\vsintegration\Vsix\VisualFSharpFull\VisualFSharpFull.csproj]
C:\Users\aviav\Source\Repos\visualfsharp\Packages\Microsoft.VSSDK.BuildTools\15.1.192\tools\vssdk\Microsoft.VsSDK.targe
ts(580,5): error MSB4018:    at Microsoft.Build.BackEnd.TaskExecutionHost.Microsoft.Build.BackEnd.ITaskExecutionHost.Ex
ecute() [C:\Users\aviav\Source\Repos\visualfsharp\vsintegration\Vsix\VisualFSharpFull\VisualFSharpFull.csproj]
C:\Users\aviav\Source\Repos\visualfsharp\Packages\Microsoft.VSSDK.BuildTools\15.1.192\tools\vssdk\Microsoft.VsSDK.targe
ts(580,5): error MSB4018:    at Microsoft.Build.BackEnd.TaskBuilder.<ExecuteInstantiatedTask>d__26.MoveNext() [C:\Users
\aviav\Source\Repos\visualfsharp\vsintegration\Vsix\VisualFSharpFull\VisualFSharpFull.csproj]
```

File: vsintegration/update-vsintegration.cmd (hunk 1 above)
Comment (brettfo) [reply vsintegration/update-vsintegration.cmd]: Eventually I will, but we have a bunch of setup stuff that's dependent upon the name `VisualFSharpFull` that I didn't want to do in this PR.

**Comment by brettfo** [comment]
What commit are you building?  Also, can you go to Help -> About in Visual Studio to check the version?  Building the latest master (commit 9639b2e8123ddc5900d04a4fa1f444b5491a9ac5) with Visual Studio Enterprise 15.6.2 I'm able to `build.cmd vs` from a clean repo.

**Comment by AviAvni** [comment]
@brettfo latest master the commit you wrote Visual Studio Enterprise  15.6.0

**Pull request merged**